src/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -223,25 +209,29 @@ protected void purchase(final boolean purchaseForBid, int PUsToSpend, final IPur
         final String stepName = step.getName();
         ProLogger.info("Simulating phase: " + stepName);
         if (stepName.endsWith("NonCombatMove")) {
+          ProData.initialize(this, dataCopy, playerCopy);
           final Map<Territory, ProTerritory> factoryMoveMap =
-              nonCombatMoveAI.doNonCombatMove(null, null, moveDel, dataCopy, playerCopy, true);
+              nonCombatMoveAI.doNonCombatMove(null, null, moveDel, true);
           if (storedFactoryMoveMap == null) {
             storedFactoryMoveMap =
-                ProSimulateTurnUtils.transferMoveMap(factoryMoveMap, unitTerritoryMap, dataCopy, data, player);
+                ProSimulateTurnUtils.transferMoveMap(factoryMoveMap, ProData.unitTerritoryMap, dataCopy, data, player);
           }
         } else if (stepName.endsWith("CombatMove") && !stepName.endsWith("AirborneCombatMove")) {
-          final Map<Territory, ProTerritory> moveMap = combatMoveAI.doCombatMove(moveDel, dataCopy, playerCopy, true);
+          ProData.initialize(this, dataCopy, playerCopy);
+          final Map<Territory, ProTerritory> moveMap = combatMoveAI.doCombatMove(moveDel, true);
```

Comment:
Would be huge to get rid of the boolean parameter on this line. As a reader of this clent code, I can almost figure out what this line is doing except for the boolean arg. For example, I do not know how the "moveMap" result value would be altered if the parameter were to be changed to false. I don't even really want to speculate on it, I would need to look at the implementation of that method. Again, as a litmus test, if you have to look up the implementation of every line of code, you're looking at something that is complex and doesn't scale. 

My suggestion, since this is another example of "a good refactor, but you can still go further", do another round of refactoring soon dedicated to removing boolean parameters. Personally I would first look to drop orphaned and duplicated code, then next I would probably reduce method argument counts, starting with boolean parameters. At that point it would then become easier to retrofit the code with tests if desired.